ts/session/messages/outgoing/visibleMessage/ClosedGroupVisibleMessage.ts

```diff
@@ -2,11 +2,14 @@ import { SignalService } from '../../../../protobuf';
 import { PubKey } from '../../../types';
 import { StringUtils } from '../../../utils';
 import { VisibleMessage } from './VisibleMessage';
-import { ClosedGroupMessage } from '../controlMessage/group/ClosedGroupMessage';
+import {
+  ClosedGroupMessage,
+  ClosedGroupMessageParams,
+} from '../controlMessage/group/ClosedGroupMessage';
 
-interface ClosedGroupVisibleMessageParams {
-  identifier?: string;
-  groupId: string | PubKey;
+interface ClosedGroupVisibleMessageParams extends ClosedGroupMessageParams {
+  // TODO Do we need strings?
```

Comment:
do we?

Reply:
So the answer to this is yes for this PR. There are a couple places where we use it as a string and it would require quite a bit of refactoring that should be done separately. See `ts/session/group/closed-group.ts` for more context. For now I'll update this comment to say refactor to PubKey only in future.

Reply:
We want it to be the `PubKey` only because the constructor verifies the string against the regex.